lib/ansible/modules/cloud/amazon/ec2_vpc_net.py

```diff
@@ -231,6 +243,11 @@ def update_vpc_tags(connection, module, vpc_id, tags, name):
                     delay=1, tries=5,
                     catch_extra_error_codes=['InvalidVpcID.NotFound'],
                 )(connection.create_tags)(Resources=[vpc_id], Tags=tags)
+
+                expected_tags = boto3_tag_list_to_ansible_dict(tags)
+                filters = [{'Name': 'tag:{0}'.format(key), 'Values': [value]} for key, value in expected_tags.items()]
+                connection.get_waiter('vpc_available').wait(VpcIds=[vpc_id], Filters=filters)
```

Comment:
Why do we need the filters here? It is to wait until the tags have actually been applied?

Reply:
Yes, that's what is happening here. I'll add a comment so it's more apparent.